frontend/app/services/MemberService.scala

```diff
@@ -390,46 +405,53 @@ class MemberService(identityService: IdentityService,
                                       joinData: PaidMemberForm,
                                       nameData: NameForm,
                                       tier: PaidTier,
-                                      customer: Stripe.Customer,
+                                      stripeCustomer: Option[Customer],
                                       campaignCode: Option[CampaignCode],
                                       email: String,
                                       ipAddress: Option[InetAddress],
                                       ipCountry: Option[Country]): Future[SubscribeResult] = {
 
     val country = joinData.zuoraAccountAddress.country
-    val planChoice = PaidPlanChoice(tier,joinData.payment.billingPeriod)
+    val planChoice = PaidPlanChoice(tier, joinData.payment.billingPeriod)
     val subscribe = zuoraService.getFeatures.map { features =>
 
       val planId = planChoice.productRatePlanId
       val plan = RatePlan(planId.get, None, featuresPerTier(features)(planId, joinData.featureChoice).map(_.id.get))
       val currency = catalog.unsafeFindPaid(planId).currencyOrGBP(joinData.zuoraAccountAddress.country.getOrElse(UK))
 
-      Subscribe(
-        account = Account.stripe(contactId = contactId, currency = currency, autopay = true),
-        paymentMethod = CreditCardReferenceTransaction(
-          cardId = customer.card.id,
-          customerId = customer.id,
-          last4 = customer.card.last4,
-          cardCountry = CountryGroup.countryByCode(customer.card.country),
-          expirationMonth = customer.card.exp_month,
-          expirationYear = customer.card.exp_year,
-          cardType = customer.card.`type`
-        ).some,
+
+      Subscribe(account = createAccount(contactId, currency, joinData.payment),
+        paymentMethod = createPaymentMethod(contactId, email, joinData.payment, stripeCustomer),
         address = joinData.zuoraAccountAddress,
         email = email,
         ratePlans = NonEmptyList(plan),
-        name = nameData,
-        ipAddress = ipAddress.map(_.getHostAddress),
-        ipCountry = ipCountry
```

Comment:
I think the changes made to drop `ipAddress` and `ipCountry` may conflict with the changes made to this file in #1401. @paulbrown1982, is that a possibility?